### PR TITLE
New README

### PR DESCRIPTION
**Pull request by mturilli**
This rewrites the README as a summary that can better serve new users both on GH and pypi

### CODE REVIEW

**Comment by mturilli** [comment]
This is now ready for review. @andre-merzky could you review the documentation imported from the (outdated and ready to be deleted) wiki pages? @mtitov could you check the binder works for you with the current `environment.yml`? For me it is a hit or miss (https://mybinder.org/v2/gh/radical-cybertools/radical.pilot/docs/readme)

File: docs/source/process/release_procedure.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+2.  create branch from latest master: e.g. ``git checkout master; git pull; git
+    checkout -b release/0.1.2``;
+3.  update version: ``echo "0.1.2" > VERSION``;
+4.  make modifications to branch: usually by merging devel ``git merge devel``
+    (make sure to pull ``devel`` before);
+5.  update version dependencies to radical stack in setup.py;
+6.  update release notes: ``$EDITOR CHANGES.md``;
+7.  commit and push: ``git commit -a; git push`` (make sure there are no
+    unwanted files in the repo);
+8.  create `pull-request
```
Comment (andre-merzky) [inline docs/source/process/release_procedure.rst:27]: We don't do release PRs for releases really.  The current release procedure is captured in this [ticket](https://github.com/radical-cybertools/radical.pilot/issues/2815) - maybe we should copy parts of that information?

Comment (mturilli) [reply docs/source/process/release_procedure.rst]: Thanks @andre-merzky . I copied over some of the text of the indicated ticket but we are still missing the `git-dsyn` script and editing the instructions for an hotfix. Could you do the needful and commit directly to this PR? I do not have the detail to do that myself. Also, you may want to update the page about testing. Also in that case, I do not have enough details to do it myself.

**Review by andre-merzky — APPROVED** [review]
I pushed small iterations and left a comment - but overall looks ready to me.

**Pull request merged**